### DISCUSSION TITLE
How to connect to Zellij server from a client in the context of a different login session

### DISCUSSION THREAD

**Discussion by crabdancing** [zellij-org/zellij#3550] [Q&A]
So, I want to help a friend out with their computer interactively using Zellij. They have a user account, let's say, `user`. They have a Zellij session open, started via the GUI, and I want to connect to them. The only problem is, when I connect over SSH, switch to the same unix account via `sudo -u user -i`, and try to `zellij attach` to their session -- it just fails and reports `No active zellij sessions found.`. What can I do to get it to work as expected?

**Comment by crabdancing** [comment]
Note that I have already tried manually setting `DBUS_SESSION_BUS_ADDRESS`, but Zellij still reports the same issue.

**Comment by giyany** [comment]
You would need to login to SSH with the same user as the one running the zellij session.

**Reply by crabdancing** [reply]
Well, obviously -- but I was already trying that from day one. It still can't find the other Zellij session. :/

**Reply by Lockszmith-GH** [reply]
From your description, you're logging in as user B and running sudo into user A - that's not the same.

If User A is runniing the zellij session.
You will need to ssh into User A and open zellij on the same session name user A is running.

**Reply by crabdancing** [reply]
So... why does it have to be the same? Is `sudo` fucking up some of the environment variables? If so, 
which ones?

**Reply by Lockszmith-GH** [reply]
I mean, probably - I've never really researched it.

Can't do it now, but I'll definitly try later (I'm intrigued).

but the way I'll test it:
* SSH into User A, before running zellij, dump the env with `env` into a file `env1`
* SSH into User B, dump the env with `env` into `env2`
* sudo from User B to User A, dump the env into `env2-sudo`

and now run some diffs, see if something makes sense.

**Reply by giyany** [reply]
This is not a topic I'm proficient in, but isn't logging in with a user mean sourcing an environment while sudo is in principle a change of permissions?

**Reply by Lockszmith-GH** [reply]
I doubt it's down to the env, but I wouldn't put money on it - that's why `env` needs to be diffed.

by default, `sudo` doesn't pass the env from the source session, unless specified to do so.
And `sudo -i` is supposed to create a new login shell, just like SSH does.
So in theory `sudo -i` should already run correctly, but the problem is zellij's ability to communicate with it's server - it's not identifying it.
